core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -978,6 +978,7 @@ class LogManager(logDirs: Seq[File],
         checkpointRecoveryOffsetsAndCleanSnapshotsInDir(logDir, logsToCheckpoint, ArrayBuffer.empty)
         checkpointLogStartOffsetsInDir(logDir, logsToCheckpoint)
       }
+      removedLog.removeLogMetrics()
```

Comment:
We use the same pattern of removing the log from `futureLogs` or `currentLogs` and then removing the corresponding log metrics in `handleLogDirFailure`. Do you think it'd be useful to turn this into a helper method and reuse the logic in both places?

Reply:
That's a good catch, thanks. I'll add that.